torchmetrics/functional/text/rouge.py

```diff
@@ -303,14 +355,17 @@ def rouge_score(
             raise ValueError(f"Got unknown rouge key {key}. Expected to be one of {list(ALLOWED_ROUGE_KEYS.keys())}")
     rouge_keys_values = [ALLOWED_ROUGE_KEYS[key] for key in rouge_keys]
 
+    if isinstance(targets, list) and bool(targets) and all(isinstance(target, str) for target in targets):
```

Comment:
Is `bool(targets` condition really necessary here? If it's required for some mypy checks, it should be sufficient to have:
```suggestion
    if isinstance(targets, list) and targets and all(isinstance(target, str) for target in targets):
```
As `targets` will be evaluated `False` if it's empty and `True` otherwise.

Reply:
@stancld Thanks for pointing this out. It shouldn't be necessary, I guess. I was trying some variations of the if clause but forgot to remove it. Let me edit and test it once before committing. Although, if necessary I would prefer `bool(targets)` over `targets` for clarity.